lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -208,19 +223,72 @@ def onCreate(self, *args):
-                cmds.menuItem( label="Copy Attribute Value",   command=lambda *args: setClipboardData(str(values)) )
-                cmds.menuItem( label="Print to Script Editor", command=lambda *args: MGlobal.displayInfo(str(values)) )
+            attrLabel = getPrettyName(self.attrName) if self.useNiceName else self.attrName
+            singleWidgetWidth = mel.eval('global int $gAttributeEditorTemplateSingleWidgetWidth; $gAttributeEditorTemplateSingleWidgetWidth += 0')
+            with AEUITemplate():
+                # See comment in ConnectionsCustomControl below for why nc=5.
+                rl = cmds.rowLayout(nc=5, adj=3)
+                with LayoutManager(rl):
+                    cmds.text(nameTxt, al='right', label=attrLabel, annotation=attr.GetDocumentation())
+                    cmds.textField(attrTypeFld, editable=False, text=typeNameStr, width=singleWidgetWidth*1.5)
+
+                    if hasValue:
+                        cmds.popupMenu()
+                        cmds.menuItem( label="Copy Attribute Value",   command=lambda *args: setClipboardData(str(values)) )
+                        cmds.menuItem( label="Print to Script Editor", command=lambda *args: MGlobal.displayInfo(str(values)) )
         else:
             cmds.error(self.attrName + " must be an array!")
 
     def onReplace(self, *args):
         pass
 
+def showEditorForUSDPrim(usdPrimPathStr):
+    # Simple helper to open the AE on input prim.
+    mel.eval('evalDeferred "showEditor(\\\"%s\\\")"' % usdPrimPathStr)
+
+# Custom control for all attributes that have connections.
+class ConnectionsCustomControl(object):
+    def __init__(self, ufeItem, prim, attrName, useNiceName):
+        self.path = ufeItem.path()
+        self.prim = prim
+        self.attrName = attrName
+        self.useNiceName = useNiceName
+        super(ConnectionsCustomControl, self).__init__()
+
+    def onCreate(self, *args):
+        frontPath = self.path.popSegment()
+        attr = self.prim.GetAttribute(self.attrName)
+        attrLabel = getPrettyName(self.attrName) if self.useNiceName else self.attrName
+        attrType = attr.GetMetadata('typeName')
+
+        singleWidgetWidth = mel.eval('global int $gAttributeEditorTemplateSingleWidgetWidth; $gAttributeEditorTemplateSingleWidgetWidth += 0')
+        with AEUITemplate():
+            # Because of the way the Maya AE template is defined we use a 5 column setup, even
+            # though we only have two fields. We resize the main field and purposely set the
+            # adjustable column to 3 (one we don't have a field in). We want the textField to
+            # remain at a given width.
+            rl = cmds.rowLayout(nc=5, adj=3)
+            with LayoutManager(rl):
+                cmds.text(nameTxt, al='right', label=attrLabel, annotation=attr.GetDocumentation())
+                cmds.textField(attrTypeFld, editable=False, text=attrType, backgroundColor=[0.945, 0.945, 0.647], width=singleWidgetWidth*1.5)
```

Comment:
Same logic/sizing as array attributes (above). The text field is the same 1.5 times for consistency. Also set the yellow background color (like Maya). And non-editable.